PWGHF/TableProducer/treeCreatorDsToKKPi.cxx

```diff
@@ -0,0 +1,521 @@
+  Configurable<bool> fillOnlyBackground{"fillOnlyBackground", false, "Flag to fill derived tables with background for ML trainings"};
+  Configurable<float> downSampleBkgFactor{"downSampleBkgFactor", 1., "Fraction of background candidates to keep for ML trainings"};
+  Configurable<float> ptMaxForDownSample{"ptMaxForDownSample", 10., "Maximum pt for the application of the downsampling factor"};
+
+  Filter filterSelectCandidates = aod::hf_sel_candidate_ds::isSelDsToKKPi >= selectionFlagDs || aod::hf_sel_candidate_ds::isSelDsToPiKK >= selectionFlagDs;
+  Filter filterMcGenMatching = nabs(o2::aod::hf_cand_3prong::flagMcMatchGen) == static_cast<int8_t>(BIT(DecayType::DsToKKPi)) && aod::hf_cand_3prong::flagMcDecayChanGen == decayChannel;
+
+  using candDsData = soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelDsToKKPi>>;
+  using candDsMcReco = soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelDsToKKPi, aod::HfCand3ProngMcRec>>;
+  using candDsMcGen = soa::Filtered<soa::Join<aod::McParticles, aod::HfCand3ProngMcGen>>;
+
+  Partition<candDsData> selectedDsToKKPiCand = aod::hf_sel_candidate_ds::isSelDsToKKPi >= selectionFlagDs;
+  Partition<candDsData> selectedDsToPiKKCand = aod::hf_sel_candidate_ds::isSelDsToPiKK >= selectionFlagDs;
+
+  Partition<candDsMcReco> recSig = nabs(aod::hf_cand_3prong::flagMcMatchRec) == static_cast<int8_t>(BIT(DecayType::DsToKKPi)) && aod::hf_cand_3prong::flagMcDecayChanRec == decayChannel;
+  Partition<candDsMcReco> recBkg = nabs(aod::hf_cand_3prong::flagMcMatchRec) != static_cast<int8_t>(BIT(DecayType::DsToKKPi));
+
+  void init(InitContext const&)
+  {
+  }
+
+  template <typename T>
+  void fillEvent(const T& collision, int isEventReject, int runNumber)
+  {
+    rowCandidateFullEvents(
+      collision.bcId(),
+      collision.numContrib(),
+      collision.posX(),
+      collision.posY(),
+      collision.posZ(),
+      isEventReject,
+      runNumber);
+  }
+
+  /// Fill table accounting for MC information and mass hypothesis
+  /// \param doMc true to fill MC information
+  /// \param massHypo mass hypothesis considered: 0 = KKPi, 1 = PiKK
+  /// \param candidate is candidate
+  template <bool doMc = false, int massHypo = 0, typename T>
+  void fillCandidateTable(const T& candidate)
+  {
+    int8_t flagMc = 0;
+    int8_t originMc = 0;
+    if constexpr (doMc) {
+      flagMc = candidate.flagMcMatchRec();
+      originMc = candidate.originMcRec();
+    }
+
+    float invMassDs = 0;
+    float deltaMassPhiKK = 0;
+    float absCos3PiKDs = 0;
+    if constexpr (massHypo == 0) {
+      invMassDs = invMassDsToKKPi(candidate);
+      deltaMassPhiKK = deltaMassPhiDsToKKPi(candidate);
+      absCos3PiKDs = std::abs(cos3PiKDsToKKPi(candidate));
+    } else if constexpr (massHypo == 1) {
+      invMassDs = invMassDsToPiKK(candidate);
+      deltaMassPhiKK = deltaMassPhiDsToPiKK(candidate);
+      absCos3PiKDs = std::abs(cos3PiKDsToPiKK(candidate));
+    }
```

Comment:
Why is `massHypo` a template argument instead of a function argument?

Reply:
Mostly to be consistent with the approach used to treat the MC information. The practical gains of doing this at compile time are negligible I think, also because sometime I have to check which templated function to use.
Personally, I like more this approach than having an option passed as a template argument and the other one as a function argument.